### PR TITLE
I161 import collection resources

### PR DESCRIPTION
**Pull request by ShanaLMoore**
This work allows for valkyrie collections to get imported via bulkrax with works. filesets will be handled in a different pr. 

Issue: 
- https://github.com/scientist-softserv/hykuup_knapsack/issues/161

### CODE REVIEW

File: app/factories/bulkrax/object_factory.rb [hunk 1]
```diff
@@ -104,7 +104,7 @@ def run!
 
     def update
       raise "Object doesn't exist" unless object
-      destroy_existing_files if @replace_files && ![Collection, FileSet].include?(klass)
+      destroy_existing_files if @replace_files && ![Collection, CollectionResource, FileSet].include?(klass)
```
Comment (jeremyf) [inline app/factories/bulkrax/object_factory.rb:107]: What happens when CollectionResource is not defined?  We need to favor a class method to answer the following:

`Bulkrax.config.collection_classes` and `Bulkrax.config.file_set_classes`, as we can't hard-code constants that may not exist in other non-Hyku 6 implementations.

Comment (ShanaLMoore) [reply app/factories/bulkrax/object_factory.rb:107]: ohhh yea, great point.

File: app/factories/bulkrax/valkyrie_object_factory.rb [hunk 2]
```diff
@@ -102,7 +100,7 @@ def create_work(object:, attrs:)
       perform_transaction_for(object: object, attrs: attrs) do
         transactions["work_resource.create_with_bulk_behavior"]
           .with_step_args(
-            "work_resource.add_to_parent" => { parent_id: attrs[:parent_id], user: @user },
+            "work_resource.add_to_parent" => { parent_id: attrs[related_parents_parsed_mapping], user: @user },
```
[the remark this answers is not in the document]
Comment (ShanaLMoore) [reply app/factories/bulkrax/valkyrie_object_factory.rb:103]: confirmed that this works. 
[collection-works copy.csv](https://github.com/samvera/bulkrax/files/14589684/collection-works.copy.csv)

File: app/factories/bulkrax/object_factory.rb (hunk 1 above)
Comment (ShanaLMoore) [reply app/factories/bulkrax/object_factory.rb:107]: this should be resolved in main with your changes now.

**Pull request merged**